daemon/execdriver/mount_linux.go

```diff
@@ -0,0 +1,81 @@
+// +build !windows
+
+package execdriver
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+// Mount contains information for a mount operation.
+type Mount struct {
+	Source      string `json:"source"`
+	Destination string `json:"destination"`
+	Writable    bool   `json:"writable"`
+	Private     bool   `json:"private"`
+	Slave       bool   `json:"slave"`
+}
+
+// Config creates a configs.Mount from a mount struct
+func (m *Mount) Config(c *Command) *configs.Mount {
+	if m.Source == "tmpfs" {
+		dest := filepath.Join(c.Rootfs, m.Destination)
```

Comment:
There is no point to proceed without `tar` binary, I think you should check it here.